packages/fetch-node-details/src/nodeDetailManager.ts

```diff
@@ -38,9 +42,14 @@ class NodeDetailManager {
 
   private network: TORUS_NETWORK_TYPE;
 
-  constructor({ network = TORUS_SAPPHIRE_NETWORK.SAPPHIRE_MAINNET, fndServerEndpoint, enableLogging = false }: NodeDetailManagerParams = {}) {
+  constructor({
+    network = TORUS_SAPPHIRE_NETWORK.SAPPHIRE_MAINNET,
+    keyType = KEY_TYPE.SECP256K1,
```

Comment:
Not sure if we should actually tie an instance to a specific key type. Currently, the only differentiation we have is for tss servers. My thinking was we could simply return the frost server addresses in addition to the dkls server addresses.

Also, you could have a situation where you want secp endpoints and ed endpoints. Now you would have to maintain two different instances of fnd in one client application?

Reply:
Moreover, if this keyType only refers to tss, I think it would be more intuitive to name it `tssKeyType`?

Reply:
i think keeping it generic as `keyType` gives us flexiblity to add any new end points specific to curve. Also if we expose a new field called `tssFrostEndpoints` rather than modifying actual endpoints, we have to add if and else statements in sdks where we are using it. With current approach, just pass keyType in constructor and rest of the code doesnt needs to be changed.